### DISCUSSION TITLE
generate an initial migration based off my existing database?

### DISCUSSION THREAD

**Discussion by bburdette** [SeaQL/sea-orm#1049] [Q&A]
I was able to generate entities from my existing sqlite database.  Is there a way to generate an initial migration, from either the entities or from the existing database?  So that when I run that initial migration on a new sqlite file I'd produce the same tables I have now, but empty.

**Comment by tyt2y3** [comment]
Umm mysqldump or an equivalent tool is the best way in this case.

**Comment by bburdette** [comment]
Thx.  Was hoping for a way to get to seaquery ddl, to maybe allow for targeting a different database.   Sounds like the way to get there is just to write the migration myself.

**Reply by billy1624** [reply]
Hey @bburdette, unfortunately, yes, you have to write the migration script to setup the database schema. Btw, don't forget you can also dump the SQL and run it on migration, https://www.sea-ql.org/SeaORM/docs/migration/writing-migration/#raw-sql.
